### PR TITLE
Basic transaction indexer functionality

### PR DESCRIPTION
**Pull request by nnkken**

### CODE REVIEW

File: poller/poller.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+		Client:      &http.Client{},
+		LcdEndpoint: lcdEndpoint,
+	}
+
+	batchSize := 1000
+	batch := db.NewBatch(conn, batchSize)
+	dbHeight, err := db.GetLatestHeight(conn)
+	if err != nil {
+		// TODO: retry
+		logger.L.Panicw("Cannot get height from database", "error", err)
```
Comment (williamchong007) [inline poller/poller.go:83]: prefer exit with error code instead of just log?

File: cmd/serve/serve.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+		}
+		listenAddr, err := cmd.PersistentFlags().GetString("listen-addr")
+		if err != nil {
+			logger.L.Panicw("Cannot get listen address from command line parameters", "error", err)
+		}
+		lcdEndpoint, err := cmd.PersistentFlags().GetString("lcd-endpoint")
+		if err != nil {
+			logger.L.Panicw("Cannot get lcd endpoint address from command line parameters", "error", err)
+		}
+		go rest.Run(restConn, listenAddr, lcdEndpoint)
```
Comment (williamchong007) [inline cmd/serve/serve.go:39]: Would be nice if we can check the node height vs db height here, and prompt user to import, or directly trigger an import here if the diff is too huge

File: poller/poller.go (hunk 1 above)
Comment (nnkken) [reply poller/poller.go:83]: `panic` in Go is similar to `throw` in other languages, which will crash the program if not caught (`recover` in Go).

File: cmd/serve/serve.go (hunk 2 above)
Comment (nnkken) [reply cmd/serve/serve.go:39]: Prefer to reject serving if the height difference is too large, unless something like `--ignore-height-difference` is provided.
I think 3 days of height difference (expect ~5s per block) could be considered too large?

File: cmd/serve/serve.go [hunk 3]
```diff
@@ -36,12 +38,37 @@ var Command = &cobra.Command{
...
+
+		if !ignoreHeightDiff {
+			dbHeight, err := db.GetLatestHeight(restConn)
+			if err != nil {
+				logger.L.Panicw("Cannot get height from database", "error", err)
+			}
+			blockResult, err := poller.GetBlock(&ctx, 0)
+			lcdHeight := blockResult.Block.Height
+			if lcdHeight-dbHeight > 10000 {
+				logger.L.Fatalw("height difference too large, please run `import` or add --ignore-height-difference flag", "db_height", dbHeight, "lcd_height", lcdHeight)
```
Comment (williamchong007) [inline cmd/serve/serve.go:62]: Please print the actual value `10000` in the error message

**Comment by nnkken** [comment]
> Need to wait for your golang sharing for details review

Golang is so simple that I have nothing to share
(For comparison, C++ is so complicated that I don't know what should I share)

**Pull request merged**